### PR TITLE
VZV | Map | Update Austin City Council Districts Overlay

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/2558
Closes https://github.com/cityofaustin/atd-data-tech/issues/1787

This PR updates the city council overlay's source to Socrata GeoJSON and increased its opacity. This gives us the added benefit of simplifying the layer in the request using the SoQL function [simplify_preserve_topology](https://dev.socrata.com/docs/functions/simplify_preserve_topology.html). It also adds a hover (desktop) or tap (mobile) popup within the city council districts with their #. This crash point popups overrule the council district popups to avoid conflicts or multiple popups. The order of the map layers was also adjusted to have council district below the roads, road labels, and crash points. Other overlays are still above the roads but below the crash points and labels.

Last, I componentized the compass spinner as `MapCompassSpinner` since the `Map` component was getting pretty bulky, and I made `MapCrashInfoBox` more general (`MapInfoBox`) so it could handle other features besides crash points.

#### Updated layer with hover info box
Screen Shot 2020-05-14 at 3 28 21 PM

#### Updated layer with mobile popup info box
Screen Shot 2020-05-14 at 3 28 54 PM

#### Layer order for City Council Districts layer
Screen Shot 2020-05-14 at 3 29 38 PM

#### Updated layer order for other overlays
Screen Shot 2020-05-14 at 3 29 58 PM
Screen Shot 2020-05-14 at 3 29 47 PM

### CODE REVIEW

File: atd-vzv/src/views/map/Map.js [hunk 1]
```diff
@@ -150,7 +109,7 @@ const Map = () => {
 
   // Fetch City Council Districts geojson
   useEffect(() => {
-    const overlayUrl = `https://services.arcgis.com/0L95CJ0VTaxqcmED/ArcGIS/rest/services/BOUNDARIES_single_member_districts/FeatureServer/0/query?where=COUNCIL_DISTRICT%20%3E=%200&f=geojson`;
+    const overlayUrl = `https://data.austintexas.gov/resource/7yq5-3tm4.geojson?$select=simplify_preserve_topology(the_geom,0.00001),council_district`;
```
Comment (mateoclarke) [inline atd-vzv/src/views/map/Map.js:112]: amazing. i saw the image comparison you sent John, very cool.

File: atd-vzv/src/views/map/Map.js [hunk 2]
```diff
@@ -257,30 +233,20 @@ const Map = () => {
       {buildHighInjuryLayer(overlay)}
       {!!cityCouncilOverlay && overlay.name === "cityCouncil" && (
         <Source type="geojson" data={cityCouncilOverlay}>
-          {/* Add beforeId to render beneath crash points */}
-          <Layer beforeId="base-layer" {...cityCouncilDataLayer} />
+          {/* Add beforeId to render beneath crash points, road layer, and map labels */}
+          <Layer beforeId="road-street" {...cityCouncilDataLayer} />
```
Comment (mateoclarke) [inline atd-vzv/src/views/map/Map.js:237]: really cool that you were able to get this to overlay placed nicely underneath the street layer. Its a great improvement visually.

File: atd-vzv/src/views/map/Map.js [hunk 3]
```diff
@@ -257,30 +233,20 @@ const Map = () => {
...
-          <FontAwesomeIcon icon={faCircle} color={colors.infoDark} size="4x" />
-          <FontAwesomeIcon
-            className="needle"
-            icon={faCompass}
-            color={colors.dark}
-            size="4x"
-          />
-        </StyledMapSpinner>
-      )}
+      <MapCompassSpinner isSpinning={isMapDataLoading} />
```
Comment (mateoclarke) [inline atd-vzv/src/views/map/Map.js:249]: love it when a new stateless functional component is born.

File: atd-vzv/src/views/map/Map.js (hunk 2 above)
Comment (mddilley) [reply atd-vzv/src/views/map/Map.js:237]: Thanks - I found all the layers and their IDs hidden away in the map instance. 👍

File: atd-vzv/src/views/map/Map.js (hunk 3 above)
Comment (mddilley) [reply atd-vzv/src/views/map/Map.js:249]: Me too! ⚛️👶🍼

**Pull request merged**